### PR TITLE
#4485 Send slow Page Editor times to rollbar

### PR DESCRIPTION
**Pull request by BALEHOK**
## What does this PR do?

- Part of #4485 
- Sending performance metrics from Page Editor to Rollbar

## Demo

Rollbar event
Screenshot 2022-11-02 at 5 19 41 PM

## Checklist

- [ ] Add tests
- [ ] Run Storybook and manually confirm that all stories are working
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/background/logging.ts [hunk 1]
```diff
@@ -312,6 +312,27 @@ export async function recordError(
   }
 }
 
+export async function recordWarning(
```
Comment (BALEHOK) [inline src/background/logging.ts:315]: Based on [recordError](https://github.com/pixiebrix/pixiebrix-extension/pull/4587/files#diff-18c5d896ecb7f366bb3e59350c17f10168a2606d32a8dcce4880f7a9a1afbfe2R259) and [recordLog](https://github.com/pixiebrix/pixiebrix-extension/pull/4587/files#diff-18c5d896ecb7f366bb3e59350c17f10168a2606d32a8dcce4880f7a9a1afbfe2R336)

Comment (BLoe) [reply src/background/logging.ts:315]: fyi - links like this don't work if the lines are outside the default collapsed view of the PR diff

**Review by BLoe — APPROVED** [review]

File: src/utils/performance.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * Measures the time between the app start and the current time
+ * @param action String message to log. A pattern to follow is <Place of measurement>:<Action>. For example, "sidebarExpanded:allRecipesLoaded"
+ * @param reportThreshold The threshold in milliseconds to report the measurement as warning to rollbar. Pass null to not report
+ */
+export function measurePerformanceFromAppStart(
+  action: string,
+  reportThreshold: number | null = 5000
+) {
+  const { duration } = performance.measure(APP_START_ACTION);
```
Comment (fregante) [inline src/utils/performance.ts:37]: Can this API be called multiple times with the same ID? A plain `Date.now()` comparison might work just as well if we intend to have multiple measurements like the function signature suggests

File: src/background/logging.ts (hunk 1 above)
Comment (BALEHOK) [reply src/background/logging.ts:315]: ah, my apologies

File: src/utils/performance.ts (hunk 2 above)
Comment (fregante) [reply src/utils/performance.ts:37]: `console.timeEnd` cannot be called twice with the same ID, I assumed that was the case for `performanceMeasure` as well, hence the question. 

I tested it now and it does work though 👍

Comment (BALEHOK) [reply src/utils/performance.ts:37]: got it.
and thanks for the review ;)

**Pull request merged**